Code/EnginePlugins/PhysXPlugin/Components/Implementation/PxDynamicActorComponent.cpp

```diff
@@ -428,6 +451,23 @@ void ezPxDynamicActorComponent::AddAngularImpulse(const ezVec3& vImpulse)
   }
 }
 
+void ezPxDynamicActorComponent::SetLockingFlags(ezBitflags<ezPxActorLockingFlags> flags)
+{
+  m_LockingFlags = flags;
+
+  if (m_pActor != nullptr)
+  {
+    EZ_PX_WRITE_LOCK(*(m_pActor->getScene()));
+
+    m_pActor->setRigidDynamicLockFlag(PxRigidDynamicLockFlag::eLOCK_LINEAR_X, m_LockingFlags.IsSet(ezPxActorLockingFlags::FreezePositionX));

```

Comment:
There is setRigidDynamicLockFlags() which takes all flags in one go. If we mirror the flag values in our enum we could just cast it and do one function call.

Reply:
done